ql/processes/blackscholesprocess.cpp

```diff
@@ -69,25 +69,58 @@ namespace QuantLib {
         return x0 * std::exp(dx);
     }
 
-    Real GeneralizedBlackScholesProcess::expectation(Time,
-                                                     Real,
-                                                     Time) const {
-        QL_FAIL("not implemented");
+    Real GeneralizedBlackScholesProcess::expectation(Time t0,
+                                                     Real x0,
+                                                     Time dt) const {
+        localVolatility(); // trigger update
+        if(isStrikeIndependent_) {
+            // exact value for curves
+            return x0 *
```

Comment:
Is this E[x], or exp(E[log x])? (The ambiguity was the reason we didn't implement this method.)